### PR TITLE
SER-986 | make username with colon invalid

### PR DESCRIPTION
**Pull request by idradm**
@mixth-sense @macbre @Wikia/platform-team 

Is this a safe change?

### CODE REVIEW

File: includes/User.php [hunk 1]
```diff
@@ -680,6 +680,7 @@ public static function isValidUserName( $name ) {
 		if ( $name == ''
```
Comment (WikiaTech) [inline includes/User.php:680]: MAJOR Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 115 issues
* ![MAJOR][MAJOR] 66 major
* ![MINOR][MINOR] 7 minor
* ![INFO][INFO] 42 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [UserSignupTest.php#L107](https://github.com/Wikia/app/blob/cf3c954ab6868c5be1317e5dabafa7eef1428bba/extensions/wikia/UserLogin/tests/UserSignupTest.php#L107): This function "signupDataProvider" has 205 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L440](https://github.com/Wikia/app/blob/cf3c954ab6868c5be1317e5dabafa7eef1428bba/extensions/wikia/UserLogin/tests/UserSignupTest.php#L440): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L484](https://github.com/Wikia/app/blob/cf3c954ab6868c5be1317e5dabafa7eef1428bba/extensions/wikia/UserLogin/tests/UserSignupTest.php#L484): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L494](https://github.com/Wikia/app/blob/cf3c954ab6868c5be1317e5dabafa7eef1428bba/extensions/wikia/UserLogin/tests/UserSignupTest.php#L494): This function has 14 parameters, which is greater than the 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS107)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L546](https://github.com/Wikia/app/blob/cf3c954ab6868c5be1317e5dabafa7eef1428bba/extensions/wikia/UserLogin/tests/UserSignupTest.php#L546): This function "sendConfirmationEmailDataProvider" has 280 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L625](https://github.com/Wikia/app/blob/cf3c954ab6868c5be1317e5dabafa7eef1428bba/extensions/wikia/UserLogin/tests/UserSignupTest.php#L625): Remove this unused "$mockMessagesMap101" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L630](https://github.com/Wikia/app/blob/cf3c954ab6868c5be1317e5dabafa7eef1428bba/extensions/wikia/UserLogin/tests/UserSignupTest.php#L630): Remove this unused "$expMsg101" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L638](https://github.com/Wikia/app/blob/cf3c954ab6868c5be1317e5dabafa7eef1428bba/extensions/wikia/UserLogin/tests/UserSignupTest.php#L638): Remove this unused "$mockMessagesMap102" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L643](https://github.com/Wikia/app/blob/cf3c954ab6868c5be1317e5dabafa7eef1428bba/extensions/wikia/UserLogin/tests/UserSignupTest.php#L643): Remove this unused "$expMsg102" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [User.php](https://github.com/Wikia/app/blob/cf3c954ab6868c5be1317e5dabafa7eef1428bba/includes/User.php): File "User.php" has 4703 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS104)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by onei** [comment]
@idradm just passing by, but do you mean ; instead of : (semicolon instead of colon) or vice versa?

**Comment by idradm** [comment]
The char is fine, the name is wrong. ;)

File: extensions/wikia/UserLogin/tests/UserSignupTest.php [hunk 2]
```diff
@@ -301,6 +305,8 @@ public function signupDataProvider() {
 				array( $reqParams16, $mockUserParams1, $mockUserLoginForm1, 'error', $expMsg16, $expErrParam16 ),
 				'error - not empty fake password' =>
 				array( $reqParams17, $mockUserParams1, $mockUserLoginForm1, 'error', $expMsg16, $expErrParam16 ),
+				'error - invalid user name with semicolon' =>
```
Comment (pwojcik86) [inline extensions/wikia/UserLogin/tests/UserSignupTest.php]: it should be "colon"

File: includes/User.php [hunk 3]
```diff
@@ -680,6 +680,7 @@ public static function isValidUserName( $name ) {
...
 			|| User::isIP( $name )
 			|| strpos( $name, '/' ) !== false
+			|| strpos( $name, ':') !== false
```
Comment (pwojcik86) [inline includes/User.php:683]: The debug message below needs to be updated

**Review by pwojcik86 — CHANGES_REQUESTED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 115 issues
* ![MAJOR][MAJOR] 66 major
* ![MINOR][MINOR] 7 minor
* ![INFO][INFO] 42 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [UserSignupTest.php#L107](https://github.com/Wikia/app/blob/b7fc07c0f644e353858310813be3843e18c5782e/extensions/wikia/UserLogin/tests/UserSignupTest.php#L107): This function "signupDataProvider" has 205 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L440](https://github.com/Wikia/app/blob/b7fc07c0f644e353858310813be3843e18c5782e/extensions/wikia/UserLogin/tests/UserSignupTest.php#L440): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L484](https://github.com/Wikia/app/blob/b7fc07c0f644e353858310813be3843e18c5782e/extensions/wikia/UserLogin/tests/UserSignupTest.php#L484): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L494](https://github.com/Wikia/app/blob/b7fc07c0f644e353858310813be3843e18c5782e/extensions/wikia/UserLogin/tests/UserSignupTest.php#L494): This function has 14 parameters, which is greater than the 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS107)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L546](https://github.com/Wikia/app/blob/b7fc07c0f644e353858310813be3843e18c5782e/extensions/wikia/UserLogin/tests/UserSignupTest.php#L546): This function "sendConfirmationEmailDataProvider" has 280 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L625](https://github.com/Wikia/app/blob/b7fc07c0f644e353858310813be3843e18c5782e/extensions/wikia/UserLogin/tests/UserSignupTest.php#L625): Remove this unused "$mockMessagesMap101" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L630](https://github.com/Wikia/app/blob/b7fc07c0f644e353858310813be3843e18c5782e/extensions/wikia/UserLogin/tests/UserSignupTest.php#L630): Remove this unused "$expMsg101" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L638](https://github.com/Wikia/app/blob/b7fc07c0f644e353858310813be3843e18c5782e/extensions/wikia/UserLogin/tests/UserSignupTest.php#L638): Remove this unused "$mockMessagesMap102" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [UserSignupTest.php#L643](https://github.com/Wikia/app/blob/b7fc07c0f644e353858310813be3843e18c5782e/extensions/wikia/UserLogin/tests/UserSignupTest.php#L643): Remove this unused "$expMsg102" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [User.php](https://github.com/Wikia/app/blob/b7fc07c0f644e353858310813be3843e18c5782e/includes/User.php): File "User.php" has 4703 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS104)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by pwojcik86 — APPROVED** [review]

**Pull request merged**